### ISSUE TITLE
[influxdb2] Install fails on influx setup

### ISSUE THREAD

**Issue by mproffitt** [influxdata/helm-charts#227]
I'm currently using the influxdb2 chart with Terraform and a AWS persistent volume

When upgrading/re-installing the chart, whilst maintaining the existing installation volume data, I am seeing the install fail with:

```
Error: failed post-install: job failed: DeadlineExceeded

  on ../common/helm/influxdb.tf line 1, in resource "helm_release" "influxdb":
   1: resource helm_release influxdb {
```

The failing step comes down to the execution of job-setup-admin `influx setup` task which returns 1 if the instance has already been configured. This causes the container used by the `job.setup-admin` to fail even though there isn't really an error.

```
root@test-host-influxdb2-0:/# influx setup -f --host http://test-host-influxdb2:8086 -o influxdata -b default -u admin -p ${INFLUXDB_PASSWORD} -t ${INFLUXDB_TOKEN}
Error: instance at "http://test-host-influxdb2:8086" has already been setup
See 'influx setup -h' for help
root@test-host-influxdb2-0:/# echo $?
1
```
Temporarily adding `|| echo` to the end of the influx setup command allows the chart to install correctly, however this is neither desirable, or sustainable

A more suitable solution would be to detect if influx has already been configured, in which case allow the job container to succeed, or alternatively to allow the job to be skipped entirely through a values setting?

**Comment by rawkode** [comment]
I'll add an option to disable the job, would this resolve your issue @mproffitt ?

**Comment by mproffitt** [comment]
It would indeed.

I think the method I'd submitted was an auto-skip but any option to allow the job to be bypassed if a re-install is required would be suitable and I'm happy for the onus to be back on the user to understand why the job should or should not run.

**Comment by timhallinflux** [comment]
closed via #294
